### ISSUE TITLE
Modifying jsp on tomcat8 threw ConcurrenModificationException and terminated "HotSwap Dispatcher" thread

### ISSUE THREAD

**Issue by nirav7715** [HotswapProjects/HotswapAgent#155]
HOTSWAP AGENT: 23:46:02.397 INFO (org.hotswap.agent.config.PluginRegistry) - Plugin 'org.hotswap.agent.plugin.hotswapper.HotswapperPlugin' initialized in ClassLoader 'org.apache.jasper.servlet.JasperLoader@5f9ff83d'.
Exception in thread "HotSwap Dispatcher" java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:901)
	at java.util.ArrayList$Itr.next(ArrayList.java:851)
	at org.hotswap.agent.watch.nio.EventDispatcher.callListeners(EventDispatcher.java:172)
	at org.hotswap.agent.watch.nio.EventDispatcher.run(EventDispatcher.java:125)
	at java.lang.Thread.run(Thread.java:745)

**Comment by nirav7715** [comment]
fixed #156

**Comment by NicklasWallgren** [comment]
It got the same type of error in 1.4.1.

```
Exception in thread "HotSwap Dispatcher" java.util.ConcurrentModificationException
	at java.base/java.util.ArrayList$Itr.checkForComodification(ArrayList.java:1013)
	at java.base/java.util.ArrayList$Itr.next(ArrayList.java:967)
	at org.hotswap.agent.watch.nio.EventDispatcher.run(EventDispatcher.java:124)
```
